api/apps/geoprocessing/src/export/pieces-exporters/project-custom-protected-areas.piece-exporter.ts

```diff
@@ -0,0 +1,84 @@
+import { geoprocessingConnections } from '@marxan-geoprocessing/ormconfig';
+import { ClonePiece, ExportJobInput, ExportJobOutput } from '@marxan/cloning';
+import { ResourceKind } from '@marxan/cloning/domain';
+import { FileRepository } from '@marxan/files-repository';
+import { Injectable, Logger } from '@nestjs/common';
+import { InjectEntityManager } from '@nestjs/typeorm';
+import { EntityManager } from 'typeorm';
+import {
+  ExportPieceProcessor,
+  PieceExportProvider,
+} from '../pieces/export-piece-processor';
+import { Readable } from 'stream';
+import { isLeft } from 'fp-ts/lib/Either';
+import { ClonePieceUrisResolver } from '@marxan/cloning/infrastructure/clone-piece-data';
+import { ProjectCustomProtectedAreasContent } from '@marxan/cloning/infrastructure/clone-piece-data/project-custom-protected-areas';
+
+interface ProjectCustomProtectedAreasSelectResult {
+  fullname: string;
+  ewkb: Buffer;
+}
+
+@Injectable()
+@PieceExportProvider()
+export class ProjectCustomProtectedAreasPieceExporter
+  implements ExportPieceProcessor {
+  constructor(
+    private readonly fileRepository: FileRepository,
+    @InjectEntityManager(geoprocessingConnections.default)
+    private readonly geoprocessingEntityManager: EntityManager,
+    private readonly logger: Logger,
+  ) {
+    this.logger.setContext(ProjectCustomProtectedAreasPieceExporter.name);
+  }
+
+  isSupported(piece: ClonePiece, kind: ResourceKind): boolean {
+    return (
+      piece === ClonePiece.ProjectCustomProtectedAreas &&
+      kind === ResourceKind.Project
+    );
+  }
+
+  async run(input: ExportJobInput): Promise<ExportJobOutput> {
+    console.log('hola');
+    const customProtectedAreas: ProjectCustomProtectedAreasSelectResult[] = await this.geoprocessingEntityManager.query(
```

Comment:
I think we should start using more query builders instead of `query` method of entity manager 🤔